### PR TITLE
feat: wait for server ports to come online, announce when tomcat is ready

### PR DESCRIPTION
**Pull request by ChristianMurphy**
##### Checklist

-   [x] the [individual contributor license agreement][] is signed
-   [x] commit message follows [commit guidelines][]

##### Description of change

tomcat-has-started

[individual contributor license agreement]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#individual-contributor-license-agreement
[commit guidelines]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#commit
[message properties]: https://github.com/Jasig/uPortal/tree/master/uportal-war/src/main/resources/properties/i18n
[WCAG 2.0 AA]: https://www.w3.org/WAI/WCAG20/quickref/?levels=aaa&technologies=smil%2Cpdf%2Cflash%2Csl

### CODE REVIEW

File: gradle/tasks/tomcat.gradle [hunk 1]
```diff
@@ -96,6 +98,12 @@ task tomcatStart() {
                 arg(value: 'startup.bat')
             }
         }
+
+        ant.waitfor() {
+            http(url: 'http://localhost:8080/uPortal', followRedirects: false)
+        }
+
+        announce.announce('tomcat has start', 'local')
```
Comment (ChristianMurphy) [inline gradle/tasks/tomcat.gradle]: :question: is it helpful to have an extra message specifically for tomcat started?
or is the default build complete announcement good enough?

```
{number} tasks executed
```

Comment (drewwills) [reply gradle/tasks/tomcat.gradle]: I think it _is_ helpful, but I might phrase it differently...

```
The Tomcat container has finished loading webapps and uPortal is ready for use.
```

**Comment by drewwills** [comment]
This change is a great addition -- with or without Docker.  :+1:

**Comment by drewwills** [comment]
Can this kind of test...

```
socket(server: 'localhost', port: 8887)
```

help us determine if HSQL/Tomcat is already running, and be the basis of logic that prevents trying to start one of those services if it's already up?

**Comment by drewwills** [comment]
I've tested this PR locally;  it seems to be really solid.  It's great to be able to...

```
$./gradlew hsqlStart tomcatStart portalOpen
```

and have have each of those operations begin at the earliest moment, with all of them taking about 43 seconds.

Before merging, let's...

  - Look at messages to the user
  - Look at preventing the build from starting services that are already running

**Comment by ChristianMurphy** [comment]
> Can this kind of test, help us determine if HSQL/Tomcat is already running, and be the basis of logic that prevents trying to start one of those services if it's already up?

Could it? Yes

However, from the docs at: https://docs.gradle.org/current/userguide/more_about_tasks.html#sec:skipping_tasks
It is not readily apparent how the ant `conditional` to gradle `onlyIf` connection would work.

File: gradle/tasks/tomcat.gradle (hunk 1 above)
Comment (ChristianMurphy) [reply gradle/tasks/tomcat.gradle]: Thanks @drewwills :bow:
Updated to a lifecycle log with improved wording.

**Comment by ChristianMurphy** [comment]
Thanks @drewwills,

I tested the new guards for tasks.

```
./gradlew hsqlStart

> Task :hsqlStart 
Starting embedded HSQLDB uPortal database

./gradlew hsqlStart tomcatStart

> Task :hsqlStart 
The embedded HSQLDB uPortal database is already running; nothing to do
> Task :tomcatStart 
Starting Tomcat servlet container in .gradle/tomcat
Tomcat has finished loading webapps, local uPortal server is live
```

LGTM. :white_check_mark: :+1:

**Pull request merged**